docs/en/toolkit/learning-resources/guides/dynamic_filesystem_configuration.md

```diff
@@ -0,0 +1,398 @@
+
+The Default Configuration has three of these common root variables: `@shot_root`, `@sequence_root`, and `@asset_root`. We’ll be modifying `@asset_root`, and this single change will affect every asset-related template.
+
+### Edit the template to match the path in the filesystem schema
+
+**Step 16:** Search for `asset_root` at the top of the `paths` section of `templates.yml`. In the Default Config, it looks like:
+
+`asset_root: assets/{sg_asset_type}/{Asset}/{Step}`
+
+Add `CustomEntity01` to the `asset_root` path to match the schema modifications:
+
+`asset_root: assets/{CustomEntity01}/{sg_asset_type}/{Asset}/{Step}`
+
+### Add set to the file name
+
+We’ve changed the folder structure for our files to reflect our schema changes, and now files will be read and written to the proper location. Now, let’s modify the file *name* for the Maya asset work file template, so that it also includes the set. 
+
+Find the `maya_asset_work` template definition again.  In its current state, the file *name* is 
+
+`{name}.v{version}.{maya_extension}`
+
+The `{name}` template key is a special key that represents user input in the Workfiles app’s File Save action. Let’s modify the template so that it doesn’t include any user input, and instead just consists of the current set and asset. 
+
+**Step 17:** Modify the `maya_asset_work` template definition so that it looks like this:
+
+```
+    maya_asset_work:
+        definition: '@asset_root/work/maya/{CustomEntity01}_{Asset}.v{version}.{maya_extension}'
+```
+
+This action allows you to use the Dining-Room entity proper name in the file name. The result will be something like `Dining-Room_Filet.v1.mb`.
+
+You’ve now modified `templates.yml` to reflect the new set folder in your production folder structure, and to include the name of the set in work files for asset tasks in Maya. Let’s test out the changes. 
+
+### Test it
+
+**Step 18:** Launch Maya from Shotgun Desktop.
+
+![Python app](../../../../images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/23_test_file_creation.png)
+
+In Maya, go to **Shotgun > File Open**, and in the resulting dialog, select a task on an asset for which you’ve specified a Set in Shotgun. 
+
+![Python app](../../../../images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/24_test_new_file.png)
+
+Select **+New File**.
+
+You can create a simple 3D object or just save the file using the **Shotgun > Save File**.
+
+![Python app](../../../../images/toolkit/learning-resources/guides/dynamic_filesystem_configuration/file_save.png)
+
+Success!
+
+Notice the **File Save** dialog box is displaying **Preview: Dining-Room_scene.v001.ma** using the variables that were set in the `preview:`
```

Comment:
Tiny nitpick, maybe the colon should be after the closing backtick?  Sorry if this is too minor to be worth noting!

Reply:
this actually didn't make any sense. fixed up the sentence.